backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,305 @@
+        // Assemble
+        createAndFillStepResult("one", null)
+
+        // Act
+        val actualResult = taskViewModel.checkIfNewAnswerIsSkipWhilePreviousIsNot()
+
+        // Assert
+        Assert.assertEquals(actualResult, true)
+    }
+
+    @Test
+    fun checkIfNewAnswerIsSkipWhilePreviousAnswersIsNot_ShouldReturnFalse() {
+        // Assemble
+        createAndFillStepResult("one", "tow")
+
+        // Act
+        val actualResult = taskViewModel.checkIfNewAnswerIsSkipWhilePreviousIsNot()
+
+        // Assert
+        Assert.assertEquals(actualResult, false)
+    }
+
+
+    @Test
+    fun checkIfCurrentStepIsBranchDecisionStep_ShouldReturnFalse() {
+        // Assemble
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(stepMocked)
+        `when`(taskViewModel.isReviewStep(stepMocked)).thenReturn(true)
+
+        // Act
+        val actualResult = taskViewModel.checkIfCurrentStepIsBranchDecisionStep()
+
+        // Assert
+        Assert.assertEquals(actualResult, false)
+    }
+
+    @Test
+    fun checkIfCurrentStepIsBranchDecisionStep_ShouldReturnTrue() {
+        // Assemble
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(stepMocked)
+        `when`(taskViewModel.isReviewStep(stepMocked)).thenReturn(false)
+        `when`(taskViewModel.currentTaskResult.getStepResult(any())).thenReturn(null)
+
+        // Act
+        val actualResult = taskViewModel.checkIfCurrentStepIsBranchDecisionStep()
+
+        // Assert
+        Assert.assertEquals(actualResult, true)
+    }
+
+    @Test
+    fun whenAnswersAreTheSame_ShouldShowCancelDialog() {
+        // Assemble
+        doReturn(true).`when`(taskViewModel).checkIfAnswersAreTheSame()
+
+        //act
+        taskViewModel.showCancelEditAlert()
+
+        //Assert
+        Assert.assertEquals(taskViewModel.showCancelEditDialog.value, true)
+    }
+
+    @Test
+    fun whenAnswersAreTheSame_ShouldNotShowCancelDialog() {
+        // Assemble
+        doReturn(false).`when`(taskViewModel).checkIfAnswersAreTheSame()
+
+        //act
+        taskViewModel.showCancelEditAlert()
+
+        //Assert
+        Assert.assertEquals(taskViewModel.showCancelEditDialog.value, false)
+    }
+
+    @Test
+    fun whenAnswersAreTheSame_And_CurrentStepIsBranchDecisionStep_ShouldShowSaveDialog() {
+        // Assemble
+        doReturn(true).`when`(taskViewModel).checkIfAnswersAreTheSame()
+        doReturn(true).`when`(taskViewModel).checkIfCurrentStepIsBranchDecisionStep()
+
+        //act
+        taskViewModel.checkForSaveDialog()
+
+        //Assert
+        Assert.assertEquals(taskViewModel.showSaveEditDialog.value, true)
+    }
+
+    @Test
+    fun whenAnswersAreTheSame_ShouldGoToNextStep() {
+        // Assemble
+        doReturn(false).`when`(taskViewModel).checkIfAnswersAreTheSame()
+
+        //act
+        taskViewModel.checkForSaveDialog()
+
+        //Assert
+        verify(taskViewModel).nextStep()
```

Comment:
add `verifyNoMoreInteractions(taskViewModel)` at the end.
But then you'll need to verify that `taskViewModel` is calling the things it does in the `before()` method.

Reply:
+1